lib/beaker/subcommand.rb

```diff
@@ -61,5 +63,51 @@ def provision()
         say "Hosts have already been provisioned"
       end
     end
+
+    desc 'exec', 'execute a directory, file, or beaker suite'
+    long_desc <<-LONG_DESC
+    Run a single file, directory, or beaker suite. If supplied a file or directory,
+    that resource will be run in the context of the `tests` suite; If supplied a beaker
+    suite, then just that suite will run. If no resource is supplied, then this command
+    executes the suites as they are defined in the configuration.
+    LONG_DESC
+    def exec(resource=nil)
+      if options[:help]
+        invoke :help, [], ["exec"]
```

Comment:
So in order for this to work, you need to add `option :help, :type => :boolean, :hide => true` to the subcommand metadata. Having the --help flags hidden isn't really a necessity, I think it helps to keep the help text focused on what the subcommand actually does.